README.md

```diff
@@ -9,19 +9,38 @@ The performance gains depend on the kind of program being run. For numerical ben
 # Compilation and instalation
 
 This is a modified version of the Lua interpreter so the process is the same you would use for compiling Lua. For full instructions, consult the doc/readme.html.
-
-    make linux -j4
-
+```bash
+make linux -j4
+```
 # Usage
 
 Our compiler generates a `.c` file with a `luaopen_` function. You can compile that into a `.so` module and then require it from Lua. The compilation is the same as any other extension module, except that you need to pass the path to the LuaAOT headers.
-
-    ./src/luaot test.lua -o testcompiled.c
-    gcc -shared -fPIC -O2 -I./src testcompiled.c -o testcompiled.so
-    ./src/lua -l testcompiled
-
+```bash
+./src/luaot test.lua -o testcompiled.c
+gcc -shared -fPIC -O2 -I./src testcompiled.c -o testcompiled.so
+./src/lua -l testcompiled
+```
 Be careful about the name of the generated file. If it has the same name as the original Lua file, the `require` may prioritize the Lua file instead of the compiled on. We also recommend compiling with optimizations, at least -O2.
 
+## Options
+### `-o`
+`-o` is for the output file like:
+```bash
+./src/luaot test.lua -o testcompiled.c # Compile test.lua to testcompiled.c
+```
+### `-m`
+`-m` is for what the name of luaopen function should be in the compiled file, by default it is `luaopen_<FILENAME>`.
+
+Example:
+```bash
+./src/luaot test.lua -o testcompiled.c -m luaopen_testfile # Compile test.lua to testcompiled.c with `luaopen_testfile` as the luaopen function
+```
+### `-e` 
+`-e` just tells the compiler to add a `main` symbol/function so the file can be quickly compiled to an executable like:
+```bash
+./src/luaot test.lua -o testcompiled.c -e # Compile test.lua to testcompiled.c and add a main func for compiling to executables
+gcc -o testexec testcompiled.c -llua5.4 -I/src # Compile testcompiled to an executable that will run the lua code
```

Comment:
Should the `-I/src` (absolute path) be `-I./src` (relative path)?

Reply:
I think we should link against the LuaAOT that we just compiled instead of dynamically linking against the system Lua 5.4. The following worked for me, does it work for you too?

```
gcc -c testcompiled.c -I./src
gcc -lm testcompiled.o src/liblua.a -o testexec
```

Reply:
> I think we should link against the LuaAOT that we just compiled instead of dynamically linking against the system Lua 5.4. What about this?
> 
> ```
> gcc -c testcompiled.c -I./src
> gcc -lm testcompiled.o src/liblua.a -o testexec
> ```

I don't think that is a good idea, because external c libraries that are linked with system lua5.4 might break then, for example if I `require("lfs")` which is installed in `/usr/local/share/lua/5.4`